### PR TITLE
Change username and password to config. Added option for state topic

### PR DESCRIPTION
**Pull request by Sennevds**
Added the option to modify the state topic and changed the mqtt login to use arguments instead of the environment variables

### CODE REVIEW

**Comment by Sennevds** [comment]
And now I see you want to change to a config file.
I will check how this is done in Go (I don't have that much experience in Go)

File: config.go [hunk 1]
```diff
@@ -11,12 +11,18 @@ import (
...
+	var mqttUser string
+	var mqttPassword string
+	var mqttStateTopic string
 	var sensorName string
 	var binFullTime float64
 	var unitOfMeasurement string
 	var FilePath string
 	var LoggingLevel string
 	flag.StringVar(&mqttServer, "mqtt_server", "mqtt://localhost:1883", "mqtt broker address")
+	flag.StringVar(&mqttUser, "mqtt_user", "", "mqtt user")
```
Comment (johnDorian) [inline config.go:23]: Can you modify the mqtt_user and mqtt_password flags to lookup the values from the corresponding environment variables?

```go
flag.StringVar(&mqttUser, "mqtt_user", "", "mqtt user")
flag.StringVar(&mqttPassword, "mqtt_password", "", "mqtt password")
```
```go
flag.StringVar(&mqttUser, "mqtt_user", lookUpEnv("MQTT_USER", ""), "mqtt user")
flag.StringVar(&mqttPassword, "mqtt_password", lookUpEnv("MQTT_PASSWORD", ""), "mqtt password")
```
```go
func lookUpEnv(variable, defaultVariable string) string {
	if envVariable, ok := os.LookupEnv(variable); ok {
		return envVariable
	}
	return defaultVariable
}
```

**Review by johnDorian — COMMENTED** [review]
Mostly looks good. Can you update the readme to reflect the new cmd line arguments you added.

File: config.go (hunk 1 above)
Comment (Sennevds) [reply config.go]: I've updated the repo. Is it okay like this?

Comment (johnDorian) [reply config.go]: Yeah, looks good. I will merge it and create another release for it later today. Thanks for the PR.

**Pull request merged**